rts/lra/lra-test/lra-test-basic/src/test/java/io/narayana/lra/arquillian/FailedLRAIT.java

```diff
@@ -0,0 +1,232 @@
+ * correct failure records are kept until explicitly removed.
+ */
+@RunWith(Arquillian.class)
+public class FailedLRAIT {
+
+    @ArquillianResource
+    private URL baseURL;
+
+    private Client client;
+
+    @Deployment
+    public static WebArchive deploy() {
+        return ShrinkWrap.create(WebArchive.class, FailedLRAIT.class.getSimpleName() + ".war")
+                .addPackages(true, org.codehaus.jettison.JSONSequenceTooLargeException.class.getPackage());
+    }
+
+    @Before
+    public void before() {
+        client = ClientBuilder.newClient();
+    }
+
+    @After
+    public void after() {
+        if (client != null) {
+            client.close();
+        }
+    }
+
+    /**
+     * test that a failure record is created when a participant (with status reporting) fails to compensate
+     */
+    @Test
+    public void testWithStatusCompensateFailed() throws Exception {
+        URI lra = invokeInTransaction(LRAParticipantWithStatusURI.TRANSACTIONAL_CANCEL_PATH, 500);
+
+        if (!validateStateAndRemove(lra, LRAStatus.FailedToCancel)) {
+            fail("lra not in failed list");
+        }
+    }
+
+    /**
+     * test that a failure record is created when a participant (with status reporting) fails to complete
+     */
+    @Test
+    public void testWithStatusCompleteFailed() throws Exception {
+        // invoke a method that should run with an LRA
+        URI lra = invokeInTransaction(LRAParticipantWithStatusURI.TRANSACTIONAL_CLOSE_PATH, 200);
+
+        // when the invoked method returns validate that the narayana implementation created a failure record
+        if (!validateStateAndRemove(lra, LRAStatus.FailedToClose)) {
+            fail("lra not in failed list");
+        }
+    }
+
+    /**
+     * test that a failure record is created when a participant (without status reporting) fails to compensate
+     */
+    @Test
+    public void testCompensateFailed() throws Exception {
+        URI lra = invokeInTransaction(LRAParticipantWithoutStatusURI.TRANSACTIONAL_CANCEL_PATH, 500);
+
+        if (!validateStateAndRemove(lra, LRAStatus.FailedToCancel)) {
+            fail("lra not in failed list");
+        }
+    }
+
+    /**
+     * test that a failure record is created when a participant (without status reporting) fails to complete
+     */
+    @Test
+    public void testCompleteFailed() throws Exception {
+        URI lra = invokeInTransaction(LRAParticipantWithoutStatusURI.TRANSACTIONAL_CLOSE_PATH, 200);
+
+        if (!validateStateAndRemove(lra, LRAStatus.FailedToClose)) {
+            fail("lra not in failed list");
+        }
+    }
+
+    private URI invokeInTransaction(String resourcePath, int expectedStatus) {
+        Response response = null;
+
+        try {
+            response = client.target(UriBuilder.fromUri(baseURL.toExternalForm())
+                    .path(LRAParticipantWithStatusURI.LRA_PARTICIPANT_PATH)
```

Comment:
This method is called also in tests that should test LRAParticipantWithoutStatusURI so this value can't be hardcoded if the method is to be reused.

Reply:
Hmm, that's a good point, thanks.

Reply:
Fixed with the last commit